src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import styles from './style.scss';
+
+const defaultIndex = -1;
+const baseClass = 'btn-group';
+
+class ButtonGroup extends React.Component {
+  state = {
+    currentIndex: defaultIndex
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    return {
+      currentIndex:
+        typeof props.currentIndex === 'number'
+          ? props.currentIndex
+          : state.currentIndex
+    };
+  }
+
+  handleClick = index => {
+    this.props.onIndexChange(index);
+    this.setState({ currentIndex: index });
+  };
+
+  render() {
+    const { currentIndex } = this.state;
+    const { size, fullWidth, children, className, ...restProps } = this.props;
+
+    const mappedChildren = React.Children.map(children, (child, index) =>
+      React.cloneElement(child, {
+        size,
+        fullWidth,
+        primary: false,
+        destructive: false,
+        onClick: (...args) => {
+          this.handleClick(index);
+          if (child.props.onClick) {
+            child.props.onClick(...args);
+          }
+        },
+        className: cx({
+          [className]: !!className,
+          [styles.active]: index === currentIndex
+        })
+      })
+    );
+
+    return (
+      <div role="group" className={styles[baseClass]} {...restProps}>
+        {mappedChildren}
+      </div>
+    );
+  }
+}
+
+ButtonGroup.defaultProps = {
+  onIndexChange: () => {}
```

Comment:
I wonder if `onIndexChange` shouldn't be required. What do you think?

Reply:
well actually it is not required to work If you just want to visually identify the current state and assign handlers to `onClick` prop for individual buttons